### PR TITLE
Reducing the build time

### PR DESCRIPTION
**Pull request by Markionium**
@cliffkoh As discussed per mail :)

This runs all three `tsc` processes in parallel instead of one after another.
Additionally it also runs a few tasks in parallel that are not related to eachother and do read only operations on the sourcecode. (Webpack /  Jest / TSLint)

-----------
Some initial timings from my windows machine (I'll run these on my MacBook later and add them here.

**Master**
6m 16s (Windows)
5m 56s (MacBook)

**PR Branch**
4m 13s (Windows)
4m 31s (MacBook)

### CODE REVIEW

**Comment by KevinTCoughlin** [comment]
🚀

**Comment by cliffkoh** [comment]
Let me play around with it :) 1 small concern is whether an engineer can still understand the TS error spew should there be local errors as part of the inner dev loop.

**Comment by cliffkoh** [comment]
:+1: right before your edit, I was about to suggest, in the ideal world, it swallows the output from all but one of the TS processes. I don't think I have yet to see a target-specific TS compiler error so this might work for 99% of the cases.

**Comment by dzearing** [comment]
woot make all the things faster!

**Comment by Markionium** [comment]
@dzearing @cliffkoh Sorry it took a while, but i've been Holiday the last few days ;)

I've fixed the PR up a bit and seem to have gotten pretty decent results, a 33% reduction in build time should be a decent start.

I noticed when checking out some CPU usage that we're far from using anywhere close to the max on my 6 core AMD machine, (except during the webpack steps) so we can maybe do more.

File: scripts/build.js [hunk 1]
```diff
@@ -66,4 +66,55 @@ function getPackage() {
...
+  return list.reduce((acc, tasks) => acc.concat(tasks));
+}
+
+// Filter disabled tasks if specified in the package.json.
+function removeDisabledTasks(disabledTasks = []) {
+  return tasks => tasks.filter(task => disabledTasks.indexOf(task) < 0);
+}
+
+function getTaskListAndTaskMap(process) {
+  const allTasks = [
```
Comment (bworline) [inline scripts/build.js]: **ts/tslint/jest** can run in parallel, so **tslint/jest** could be started earlier than what you have, but if **ts** finishes before **jest** (likely for the biggest project in this repo) then **webpack** is waiting when it doesn't have to.

Consider a bit more complex system than this, where you indicate each task's dependency/ies.  

sass depends on copy
ts depends on sass
tslint depends on sass
jest depends on sass
webpack depends on ts

Use this dependency tree to set up promises stored in a dictionary and indexed by task name, and you can have a more efficient build.

**Comment by bworline** [comment]
Have you reached out to the Rush people to see if they have solved the problem with running jest tests in parallel? (The "--runInBand" jest flag is forcing serialization.)

Another rush-based repo I work in within MS has been running jest tests in parallel for a while.

**Pull request merged**